elements/lisk-dpos/src/delegates_list.ts

```diff
@@ -253,26 +255,21 @@ export class DelegatesList {
 	): Promise<void> {
 		const round = this.rounds.calcRound(height) + roundOffset;
 		debug(`Creating vote weight snapshot for round: ${round.toString()}`);
-		// This list is before executing the current block in process
-		const originalDelegates = await this.chain.dataAccess.getDelegates();
-
-		// Merge updated delegate accounts
-		const updatedAccounts = stateStore.account.getUpdated();
-		const updatedAccountsMap: { [address: string]: Account } = {};
-		// Convert updated accounts to map for better search
-		for (const account of updatedAccounts) {
-			// Insert only if account is a delegate
-			if (account.username) {
-				updatedAccountsMap[account.address] = account;
-			}
-		}
-		// Inject delegate account if it doesn't exist
-		for (const delegate of originalDelegates) {
-			if (updatedAccountsMap[delegate.address] === undefined) {
-				updatedAccountsMap[delegate.address] = delegate;
-			}
-		}
-		const delegates = [...Object.values(updatedAccountsMap)];
+
+		// Data format for the registered delegates
+		// chain:usernames => { registeredDelegates: { username, address }[] }
+		const usernamesStr = await stateStore.chain.get(
+			CHAIN_STATE_DELEGATE_USERNAMES,
+		);
+		const usernames = usernamesStr
+			? (JSON.parse(usernamesStr) as ChainStateUsernames)
+			: { registeredDelegates: [] };
```

Comment:
I don't know if it's really necessary but maybe we can wrap this with try/catch and throw a better error for logging at the framework level

Reply:
We don't throw errors normally in the transactions. And its just initializing default value for the attribute, which we do a similar way across the application.

Reply:
I get it for transactions but I see that we are throwing error while getting a delegate list or verify forger and other places in this file. But just highlighting that if JSON.parse will fail then it will be hard to debug

Reply:
The function itself is `async`. I think if there will be any synchronous error it will reject the promise and top-level code will catch the error.

Reply:
also here I was highlighting `JSON.parse` instead of async task

Reply:
Almost everywhere in our code, I see `JSON.parse` wrapped around `try/catch` but if you think it will be fine and will be handled properly in case of parsing error then you can ignore it. I just wanted to highlight it.

Reply:
@ishantiw If there is any JSON which is populated by user input then we should try/catch it but in general we don't. See my comment. 

https://github.com/LiskHQ/lisk-sdk/pull/5338#discussion_r427166427

Reply:
I see your point. Maybe here its safe to assume that the string will be parsed without error